integration-tests/app_id_test.go

```diff
@@ -7,110 +7,116 @@ import (
 	"time"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 )
 
-func TestAppIdSuccess(t *testing.T) {
+type AppCredentialTestSuite struct {
+	suite.Suite
+}
+
+func (suite *AppCredentialTestSuite) SetupSuite() {
 	err := StartContainers("./configs/app-id/docker-compose.yaml")
-	// Sleep required in github actions to prevent failing of tests in case of services getting late to start.
 	time.Sleep(10 * time.Second)
 	if err != nil {
 		fmt.Printf("Error: %v", err)
-		t.Fatalf("Error starting docker-compose: %v", err)
+		suite.Errorf(err, "Error starting docker-compose: %v")
 	}
+}
 
+func TestAppCredentialSuite(t *testing.T) {
+	fmt.Println("Running AppCredentialTestSuite")
+	suite.Run(t, new(AppCredentialTestSuite))
+}
+
+func (suite *AppCredentialTestSuite) TearDownSuite() {
+	fmt.Println("Stopping AppCredentialTestSuite")
+	errStop := StopContainers("./configs/app-id/docker-compose.yaml")
+	if errStop != nil {
+		fmt.Printf("Error stopping : %v", errStop)
+	}
+}
+
+func (suite *AppCredentialTestSuite) TestAppIdSuccess() {
 	client := &http.Client{}
 	req, errReq := http.NewRequest("GET", "http://127.0.0.1:9095/", nil)
-	if errReq != nil {
-		fmt.Printf("Error: %v", errReq)
-		StopContainers("./configs/app-id/docker-compose.yaml")
-		t.Fatalf("Error creating http request: %v", errReq)
-	}
+	require.Nilf(suite.T(), errReq, "Error creating the HTTP request: %v", errReq)
 	req.Header = http.Header{
 		"Host":      []string{"localhost"},
-		"x-app-id":  []string{"fcf4db29"},
-		"x-app-key": []string{"9a0435ee68f5d647f03a80480a97a326"},
+		"x-app-id":  []string{"23f118be"},
+		"x-app-key": []string{"44d128988763aee1b0ff0691f9686f7e"},
```

Comment:
Will definitely need to read these from the environment or some file accessible from these tests.

Reply:
@unleashed Encrypted secrets option is there but didn't use it because keys may get changed during the dev stage. Once we are stable with the tests, we can add to that. But we'll need to add several app_ids, service_ids. Is there a better option than encrypted secrets ?